File: PixelAdventure/PixelAdventure/GameScene.m
//
//  GameScene.m
//  PixelAdventure
//
//  Created by Reed Carson on 6/8/15.
//  Copyright (c) 2015 Reed Carson. All rights reserved.
//

#import "GameScene.h"

@implementation GameScene

{
    
    SKShapeNode* pixel;
    
}

-(void)didMoveToView:(SKView *)view {
   
    
    
    
    self.physicsBody = [SKPhysicsBody bodyWithEdgeLoopFromRect:view.frame];
    
//    SKFieldNode* field = [SKFieldNode radialGravityField];
//    
//    field.position = view.center;
//    field.strength = 1;
//    
//    [self addChild:field];
    
    
    self.backgroundColor = [UIColor blackColor];
    
    pixel = [SKShapeNode shapeNodeWithEllipseOfSize:CGSizeMake(25, 25)];
    
    pixel.fillColor = [UIColor cyanColor];
    pixel.strokeColor = [UIColor blackColor];
    pixel.physicsBody = [SKPhysicsBody bodyWithCircleOfRadius:12.5];
    pixel.glowWidth = 2;
    pixel.strokeColor = [UIColor redColor];
    pixel.position = CGPointMake(10, 10);

//    pixel.physicsBody.friction = 150;
    [self addChild:pixel];
}

- (void)movePixelInDirection:(CGVector)direction {
    
//    [pixel.physicsBody applyImpulse:direction];
//    [pixel.physicsBody applyForce:direction];
    
    CGFloat newX = pixel.position.x + direction.dx;
    CGFloat newY = pixel.position.y + direction.dy;
    
    pixel.position = CGPointMake(newX, newY);
    
}

- (void)normalAttack {
    
    SKShapeNode* babyPixel = [SKShapeNode shapeNodeWithEllipseOfSize:CGSizeMake(10, 10)];
    
    babyPixel.fillColor = [UIColor blueColor];
    babyPixel.strokeColor = [UIColor blackColor];
    
    babyPixel.position = CGPointMake(pixel.position.x + 10, pixel.position.y + 10);
    
    [self addChild:babyPixel];
    
    babyPixel.physicsBody = [SKPhysicsBody bodyWithCircleOfRadius:5];
    
    babyPixel.physicsBody.affectedByGravity = NO;
    
    [babyPixel.physicsBody applyImpulse:CGVectorMake(10, 0)];
    
}

- (void)specialAttack {
    
    SKEmitterNode* fireball = [SKEmitterNode nodeWithFileNamed:@"attack"];
    
    fireball.position = CGPointMake(pixel.position.x + 10, pixel.position.y + 10);
    
    [self addChild:fireball];
    
    fireball.physicsBody = [SKPhysicsBody bodyWithCircleOfRadius:5];
    
    fireball.physicsBody.affectedByGravity = YES;
    
    [fireball.physicsBody applyImpulse:CGVectorMake(2.5, 2.5)];

    
}

//-(void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
    /* Called when a touch begins */
    
//    for (UITouch *touch in touches) {
//        
//        CGPoint location = [touch locationInNode:self];
//        
//        SKShapeNode* pixel = [SKShapeNode shapeNodeWithEllipseOfSize:CGSizeMake(25, 25)];
//        
//        pixel.fillColor = [UIColor cyanColor];
//        pixel.strokeColor = [UIColor blackColor];
//
//        pixel.physicsBody = [SKPhysicsBody bodyWithCircleOfRadius:12.5];
//        pixel.glowWidth = 1;
//        pixel.strokeColor = [UIColor redColor];
//        
//        pixel.position = location;
//        [self addChild:pixel];
//        
//    }


-(void)update:(CFTimeInterval)currentTime {
    /* Called before each frame is rendered */
}

@end

File: FrostByte/FrostByte/GameScene.swift
//
//  GameScene.swift
//  FrostByte
//
//  Created by Reed Carson on 6/24/15.
//  Copyright (c) 2015 Reed Carson. All rights reserved.
//

import SpriteKit

class GameScene: SKScene {
    
   
    var ballSize: CGFloat = 20
    
    var snowman: SKSpriteNode!
    
    
    override func didMoveToView(view: SKView) {
      
        snowman = childNodeWithName("snowman") as! SKSpriteNode
        
        buildLayout(stageLayout)
        
        let atlas = SKTextureAtlas(named: "snowman")
        
        var textures: [SKTexture] = []
        
//        for textureName in atlas.textureNames as! [String] {
        
        for i in 0..<atlas.textureNames.count {
        
        let texture = SKTexture(imageNamed: "snowman\(i)")
            
            println("snowman\(i)")
            
            textures.append(texture)
            
        }
        
        let action = SKAction.animateWithTextures(textures, timePerFrame: 0.1)
        
        let forever = SKAction.repeatActionForever(action)
        
        snowman.runAction(forever)
//
        
    }
    
    override func touchesEnded(touches: Set<NSObject>, withEvent event: UIEvent) {
  
        for touch in (touches as! Set<UITouch>) {
           
            let snowmanhead = SKShapeNode(ellipseOfSize: CGSizeMake(20, 20))
            
            snowmanhead.fillColor = UIColor.yellowColor()
            
         
            
            snowmanhead.physicsBody = SKPhysicsBody(circleOfRadius: 10)
            
              snowmanhead.physicsBody?.mass = 1
         
            snowmanhead.position = CGPointMake(100, 100)
            addChild(snowmanhead)
            
            snowmanhead.physicsBody?.applyImpulse(CGVectorMake(10000, 100))
            
        }
    }
    
    
    
    
    func buildLayout(layout: [Block]) {
       
        let gridSize: CGFloat = 20
        let gridMetric = (frame.width / 2) / gridSize
        
        for block in layout {
            
            let size = CGSizeMake(block.wide * gridMetric, block.tall * gridMetric)
            
            let blockNode = SKShapeNode(rectOfSize: size, cornerRadius: 4)
            
          
            let x = (block.left * gridMetric) + (size.width / 2) + (frame.width / 2)
            let y = (block.bottom * gridMetric) + (size.height / 2) + 20
            
            
            blockNode.position = CGPointMake(x, y)
            
            blockNode.strokeColor = UIColor.blackColor()
            blockNode.fillColor = UIColor.orangeColor()
            
            blockNode.physicsBody = SKPhysicsBody(rectangleOfSize: size)
            
            addChild(blockNode)
            
        }
        
        
        
    }
   
    override func update(currentTime: CFTimeInterval) {
       
        
        
        
    }
}

struct Block {
    
    var left: CGFloat!
    
    var bottom: CGFloat!
    
    var wide: CGFloat!
    
    var tall: CGFloat!
    
    
    init(left l: CGFloat, bottom b: CGFloat, wide w: CGFloat, tall t: CGFloat) {
        
        left = l
        bottom = b
        wide = w
        tall = t
        
    }
    
}

let stageLayout: [Block] = [

    Block(left: 5, bottom: 0, wide: 1, tall: 3),
    Block(left: 7, bottom: 0, wide: 1, tall: 3),
    Block(left: 10, bottom: 0, wide: 1, tall: 3),
    Block(left: 12, bottom: 0, wide: 1, tall: 3),
    
    Block(left: 6, bottom: 5, wide: 1, tall: 3),
    Block(left: 11, bottom: 5, wide: 1, tall: 3),
   
    Block(left: 5, bottom: 4, wide: 3, tall: 1),
    Block(left: 10, bottom: 4, wide: 3, tall: 1),
    
   

]










